### PR TITLE
Don't shuffle first media device

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/12435

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/third_party/blink/renderer/modules/mediastream/media_devices.cc [hunk 1]
```diff
@@ -33,7 +33,10 @@ void FarbleMediaDevices(ExecutionContext* context,
     // domain+session key.
     std::mt19937_64 prng =
         BraveSessionCache::From(*context).MakePseudoRandomGenerator();
-    std::shuffle(media_devices->begin(), media_devices->end(), prng);
+    MediaDeviceInfoVector::iterator it_begin = media_devices->begin();
+    MediaDeviceInfoVector::iterator it_end = media_devices->end();
+    if (it_begin != it_end) ++it_begin;
+    std::shuffle(it_begin, it_end, prng);
```
Comment (goodov) [inline chromium_src/third_party/blink/renderer/modules/mediastream/media_devices.cc:39]: I think we can do all of this only when `media_devices.size() > 2`, i.e. we can do an early return at the very start and then access `settings` and do all that shuffle only if we have to shuffle anything.

Comment (pilgrim-brave) [reply chromium_src/third_party/blink/renderer/modules/mediastream/media_devices.cc]: Done in https://github.com/brave/brave-core/commit/8e6154354503320256bc9581169c94a537f73149

File: chromium_src/third_party/blink/renderer/modules/mediastream/media_devices.cc [hunk 2]
```diff
@@ -26,15 +26,19 @@ namespace brave {
 void FarbleMediaDevices(ExecutionContext* context,
                         MediaDeviceInfoVector* media_devices) {
   WebContentSettingsClient* settings = GetContentSettingsClientFor(context);
```
Comment (goodov) [inline chromium_src/third_party/blink/renderer/modules/mediastream/media_devices.cc:28]: settings can be placed after the media_devices size check.

**Review by goodov — APPROVED** [review]

File: chromium_src/third_party/blink/renderer/modules/mediastream/media_devices.cc (hunk 2 above)
Comment (pilgrim-brave) [reply chromium_src/third_party/blink/renderer/modules/mediastream/media_devices.cc:28]: Missed this before I merged. Followup here: https://github.com/brave/brave-core/pull/10425

**Pull request merged**